Scoop/Controller/Ride Posting/InitialPostRideViewController.swift

```diff
@@ -0,0 +1,294 @@
+        arrivalLabel.font = labelFont
+        arrivalLabel.accessibilityLabel = "name"
+        arrivalLabel.textColor = .offBlack
+        
+        arrivalTextField.textColor = .darkGray
+        arrivalTextField.layer.borderWidth = textFieldBorderWidth
+        arrivalTextField.layer.borderColor = UIColor.textFieldBorderColor.cgColor
+        arrivalTextField.layer.cornerRadius = textFieldCornerRadius
+        arrivalTextField.font = textFieldFont
+        arrivalTextField.attributedPlaceholder = NSAttributedString(string:"Arrival location", attributes: [NSAttributedString.Key.foregroundColor: UIColor.offBlack])
+        // Sets up the icon inside the text field.
+        let iconContainer = UIView(frame: CGRect(x: 0, y: 0, width: 25, height: 15))
+        let arrivalIcon = UIImageView(frame: CGRect(x: 10, y: -2.5, width: 20, height: 20))
+        arrivalIcon.image = UIImage(named: "destinationIcon")
+        arrivalIcon.contentMode = .scaleAspectFit
+        iconContainer.addSubview(arrivalIcon)
+        
+        arrivalTextField.leftView = iconContainer;
+        arrivalTextField.leftViewMode = UITextField.ViewMode.always
+        arrivalTextField.leftViewMode = .always
+        arrivalTextField.addTarget(self, action: #selector(presentArrivalSearch), for: .touchDown)
+        view.addSubview(arrivalTextField)
+        
+        arrivalTextField.snp.makeConstraints { make in
+            make.leading.equalTo(titleLabel.snp.leading)
+            make.top.equalTo(departureTextField.snp.bottom).inset(-24)
+            make.trailing.equalToSuperview().inset(32)
+        }
+
+    }
+    
+    private func setupLabels() {
+        let labelLeading = 10
+        let labelTop = 8
+        [departureLabel, arrivalLabel].forEach { label in
+            label.font = .systemFont(ofSize: 12)
+            label.textColor = .scoopDarkGreen
+            label.backgroundColor = .white
+            label.textAlignment = .center
+            label.isHidden = true
+            view.addSubview(label)
+        }
+        
+        departureLabel.text = "Departure"
+        arrivalLabel.text = "Arrival"
+        
+        departureLabel.snp.makeConstraints { make in
+            make.top.equalTo(departureTextField).inset(-labelTop)
+            make.leading.equalTo(departureTextField).inset(labelLeading)
+            make.height.equalTo(16)
+            make.width.equalTo(65)
+        }
+        
+        arrivalLabel.snp.makeConstraints { make in
+            make.top.equalTo(arrivalTextField).inset(-labelTop)
+            make.leading.equalTo(arrivalTextField).inset(labelLeading)
+            make.height.equalTo(16)
+            make.width.equalTo(40)
+        }
+        
+    }
+
+    @objc private func studentDriverButtonToggle() {
+        if taxiButton.isSelected {
+            taxiButton.isSelected.toggle()
+            studentDriverButton.isSelected.toggle()
+        } else {
+            studentDriverButton.isSelected.toggle()
```

Comment:
`if taxiButton.isSelected {
taxiButton.isSelected.toggle()
}

studentDriverButton.isSelected.toggle()
`

Not sure what the context of this code is, but taxiButton will never be able to be toggled again if its selected = false. Unless you somewhere else set isSelected = true again in your code, it will only run once I believe